cognite/client/data_classes/hosted_extractors/jobs.py

```diff
@@ -0,0 +1,478 @@
+    prefix: Prefix | None = None
+
+    @classmethod
+    def _load_job(cls, resource: dict[str, Any]) -> ValueFormat:
+        return cls(
+            encoding=resource["encoding"],
+            compression=resource["compression"],
+            prefix=Prefix._load(resource["prefix"]) if "prefix" in resource else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case)
+        if self.prefix:
+            output["prefix"] = self.prefix.dump(camel_case)
+        return output
+
+
+@dataclass
+class RockwellFormat(JobFormat):
+    _type = "rockwell"
+    encoding: Literal["utf16", "utf16le"]
+    compression: str = "gzip"
+    prefix: Prefix | None = None
+
+    @classmethod
+    def _load_job(cls, resource: dict[str, Any]) -> RockwellFormat:
+        return cls(
+            encoding=resource["encoding"],
+            compression=resource["compression"],
+            prefix=Prefix._load(resource["prefix"]) if "prefix" in resource else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case)
+        if self.prefix:
+            output["prefix"] = self.prefix.dump(camel_case)
+        return output
+
+
+@dataclass
+class CustomFormat(JobFormat):
+    _type = "custom"
+    encoding: Literal["utf16", "utf16le"]
+    mapping_id: str
+    compression: str = "gzip"
+
+    @classmethod
+    def _load_job(cls, resource: dict[str, Any]) -> CustomFormat:
+        return cls(
+            encoding=resource["encoding"],
+            compression=resource.get("compression", "gzip"),
+            mapping_id=resource["mappingId"],
+        )
+
+
+@dataclass
+class CogniteFormat(JobFormat):
+    _type = "cognite"
+    encoding: Literal["utf16", "utf16le"]
+    compression: str = "gzip"
+    prefix: Prefix | None = None
+
+    @classmethod
+    def _load_job(cls, resource: dict[str, Any]) -> CogniteFormat:
+        return cls(
+            encoding=resource["encoding"],
+            compression=resource["compression"],
+            prefix=Prefix._load(resource["prefix"]) if "prefix" in resource else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case)
+        if self.prefix:
+            output["prefix"] = self.prefix.dump(camel_case)
+        return output
+
+
+@dataclass
+class JobConfig(CogniteObject, ABC):
+    @classmethod
+    @abstractmethod
+    def _load_config(cls, data: dict[str, Any]) -> Self:
+        raise NotImplementedError()
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        if "topicFilter" in resource:
+            return cast(Self, MQTTConfig._load_config(resource))
+        elif "topic" in resource:
+            return cast(Self, KafkaConfig._load_config(resource))
+        else:
+            return cast(Self, UnknownCogniteObject(resource))
```

Comment:
There's a third it seems:
```suggestion
        elif "interval" in resource and "path" in resource:
            return ...
        else:
            return cast(Self, UnknownCogniteObject(resource))
```